### ISSUE TITLE
Reproducing CIFAR10 Results with this Repo

### ISSUE THREAD

**Issue by robbiejones96** [google-research/simclr#95]
Hello SimCLR contributors! Thank you so much for sharing your implementation with the broader research community. I wanted to ask about some problems I've been having reproducing the CIFAR10 results reported in this repository. I've run these two commands, one after the other (taken from the README):

```
python run.py --train_mode=pretrain   --train_batch_size=512 --train_epochs=1000   --learning_rate=1.0 \
--weight_decay=1e-4 --temperature=0.5   --dataset=cifar10 --data_dir=/juice/scr/rmjones/cifar10 \
--image_size=32 --eval_split=test --resnet_depth=18   --use_blur=False --color_jitter_strength=0.5  \
--model_dir=cifar10_models/firsttry --use_tpu=False --cache_dataset=True
```
and
```
python run.py --mode=train_then_eval --train_mode=finetune   --fine_tune_after_block=4 \
--zero_init_logits_layer=True   --variable_schema='(?!global_step|(?:.*/|^)Momentum|head)'   \
--global_bn=False --optimizer=momentum --learning_rate=0.1 --weight_decay=0.0   \
--train_epochs=100 --train_batch_size=512 --warmup_epochs=0   --dataset=cifar10 \
--data_dir=/juice/scr/rmjones/cifar10 --image_size=32 --eval_split=test --resnet_depth=18 \
--checkpoint=cifar10_models/firsttry --model_dir=cifar10_models/firsttry_ft --use_tpu=False
```
The README says these two commands should give around a 91% accuracy, but the `label_top_1_accuracy` reported from my run is only 77.56%. I'm currently running with these versions of `tensorflow` (per `requirements.txt`):

```
tensorboard==1.15.0
tensorflow-datasets==3.1.0
tensorflow-estimator==1.15.1
tensorflow-gpu==1.15.2
tensorflow-hub==0.8.0
tensorflow-metadata==0.24.0
```

It would be great if I could get some pointers for any mistakes I've made or places to look to try debugging. Thank you!

**Comment by chentingpc** [comment]
Thanks for reporting the issue. The command here was for SimCLRv1 (though I tried to make it consistent for v2 as well), but can you try to run the same command using SimCLRv1 code (https://github.com/google-research/simclr/releases/tag/1.0)? 

Alternatively, you can try the following command for pretraining (with existing simclrv2 codebase) which should also give you numbers ~91%:
```
python run.py --train_mode=pretrain   --train_batch_size=258 --train_epochs=400  \
 --learning_rate=0.2 --learning_rate_scaling=sqrt --proj_out_dim=64 --num_proj_layers=2 \
--weight_decay=1e-4 --temperature=0.2   --dataset=cifar10 --data_dir=/juice/scr/rmjones/cifar10 \
--image_size=32 --eval_split=test --resnet_depth=18   --use_blur=False --color_jitter_strength=0.5  \
--model_dir=cifar10_models/firsttry --use_tpu=False --cache_dataset=True
```

**Comment by robbiejones96** [comment]
Great, I appreciate the quick response! Running the original command from the README on the v1 code that you linked does indeed give ~91% test accuracy. I am still receiving poor test accuracy on the v2 repo, however (71.25%). I ran

```
python run.py --train_mode=pretrain --train_batch_size=256 --train_epochs=400 --learning_rate=0.2 \
--learning_rate_scaling=sqrt --proj_out_dim=64 --num_proj_layers=2 --weight_decay=1e-4 --temperature=0.2 \
--dataset=cifar10 --data_dir=/juice/scr/rmjones/cifar10 --image_size=32 --eval_split=test --resnet_depth=18 \
--use_blur=False --color_jitter_strength=0.5 --model_dir=cifar10_models/v2secondtry --use_tpu=False \
--cache_dataset=True
```
and
```
python run.py --mode=train_then_eval --train_mode=finetune --fine_tune_after_block=4 \
--zero_init_logits_layer=True '--variable_schema=(?!global_step|(?:.*/|^)Momentum|head)' --global_bn=False \
--optimizer=momentum --learning_rate=0.1 --weight_decay=0.0 --train_epochs=100 --train_batch_size=512 \
--warmup_epochs=0 --dataset=cifar10 --data_dir=/juice/scr/rmjones/cifar10 --image_size=32 --eval_split=test \
--resnet_depth=18 --checkpoint=cifar10_models/v2secondtry --model_dir=cifar10_models/v2secondtry_ft \
--use_tpu=False
```

where the first command is the one you gave in your response, and the second is the same finetuning command intended for the v1 repo. Does the finetuning command also need to be adapted for the v2 architecture?

**Comment by chunhanl** [comment]
Interesting
Does this somehow means v1 is actually better than v2 on cifar10?

**Comment by chunhanl** [comment]
FYI @robbiejones96
A little follow up for this thread,
I've successfully reached **90.5% top1 accuracy** on cifar10 using

```
python run.py --train_mode=pretrain  --train_batch_size=256 --train_epochs=400  \
--learning_rate=0.2 --learning_rate_scaling=sqrt --proj_out_dim=64 --num_proj_layers=2 \
--weight_decay=1e-4 --temperature=0.2 \
--dataset=cifar10 --data_dir=/tmp/dataset \
--image_size=32 --eval_split=test --resnet_depth=18   --use_blur=False --color_jitter_strength=0.5  \
--model_dir=/tmp/simclr/pretrain_model \
--use_tpu=True --tpu_name=some-tpu --cache_dataset=True
```
```
python run.py --mode=train_then_eval --train_mode=finetune --train_batch_size=512 --train_epochs=100 \
--fine_tune_after_block=4 --zero_init_logits_layer=True '--variable_schema=(?!global_step|(?:.*/|^)Momentum|head)' --global_bn=False \
--optimizer=momentum --learning_rate=0.1 --weight_decay=0.0 --warmup_epochs=0 \
--dataset=cifar10 --data_dir=/tmp/dataset \
--image_size=32 --eval_split=test --resnet_depth=18 \
--checkpoint=/tmp/simclr/pretrain_model \
--model_dir=/tmp/simclr/finetune_model \
--use_tpu=True --tpu_name=some-tpu --cache_dataset=True
```
on **simclrv2 code base**.
Hope this helps.

**Comment by robbiejones96** [comment]
@chunhanl Cool! Thank you for sharing this. I'm going to close this issue for now, as I've decided to move forward with the SimCLR v1 implementation. If I end up coming back to the v2 code base, I'll definitely use your hyperparams as a starter. Thanks again!
